src/lib/dnssd/minimal_mdns/RecordData.cpp

```diff
@@ -36,25 +37,30 @@ bool ParseTxtRecord(const BytesRange & data, TxtRecordDelegate * callback)
         {
             return false;
         }
-        pos++;
 
         // name=value string of size length
-        const uint8_t * equalPos = pos;
-        while ((*equalPos != '=') && ((equalPos - pos) < length))
+        const uint8_t * equalPos = pos + 1;
```

Comment:
how about doing a `pos++; // skip to the first valid character` call here and modify less below? It seems code updated everywhere to be pos+1.

Reply:
Can (should?) also update a unit test or add one that uses heap instead of stack for the packet buffer and that can let ASAN/valgrind validate offsets for us, so we discover these during unit tests instead of at runtime.